src/main/java/ee/tuleva/onboarding/user/UserService.java

```diff
@@ -82,4 +87,9 @@ public User save(User user) {
     log.info("Saving user: userId={}", user.getId());
     return userRepository.save(user);
   }
+
+  public boolean isEmailExist(String personalCode, String email) {
+    Optional<User> existingUser = userRepository.findByEmail(email);
+    return existingUser.isPresent() && !personalCode.equals(existingUser.get().getPersonalCode());
+  }
```

Comment:
Why is this a public method? I.e. does it necessarily need to be as part of the public interface of this class or can it be private? Is it used anywhere outside of this class? Should it ever be used outside of the `UserService`?

Reply:
Also, the name of this method is not grammatically correct in English. I usually like to Google phrases to see how many responses I get: https://www.google.com/search?q=%22is+email+exist%22 shows 10 responses.
I would name it something like "isExistingEmail" or "isDuplicateEmail"

Reply:
Oh, it's used in the `UserDetailsUpdater`, got it.

Reply:
Yeah, the mandatory Java **is** prefix always creates interesting situations, since we can't use **does**, so in my past lives I've just ignored the grammar. I'll change it to something more sensible.